design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+
+The key aspect of a stack is that it will be installed with permissions to only watch for resources **within the namespace** it is running in.
+This provides a nice isolation as well as versioning story.
+
+* **Isolation:** The stack should not be able to read or write any resources outside of its target namespace.
+* **Versioning:** Multiple versions of the stack can be installed within the control plane, each namespace can have their own independent version.
+
+### Installation Flow
+
+Let's look at a concrete example of what happens when the `WordPress` stack is installed into Crossplane in a target namespace of the user's choice:
+
+* The user creates an `StackInstall` and specifies the WordPress stack
+* All CRDs defined by the WordPress stack will be installed into the cluster (remember that CRDs themselves are always cluster scoped).
+In the case of WordPress, a single CRD is installed that represents a WordPress application instance, `WordPressInstance`.
+* A `Role` will be created in the target namespace that includes the RBAC rules for the [allowed core Kubernetes kinds](#allowed-resource-access), the stack's owned CRDs, and its dependent CRDs.
+Because this is a `Role`, this only grants the given permissions to namespaced resources in the target namespace.
+* A `ServiceAccount` is created in the namespace that the stack's controller will be running that will serve as the runtime identity of the controller.
+* A `RoleBinding` is created in the target namespace to bind the namespaced `Role` and the namespaced `ServiceAccount`.
+* A `Deployment` is created in the same namespace as the stack and it is expected to launch and manage the pod that hosts the WordPress controller.
+
+### Architecture Diagram - Stack
+
+The diagram below summarizes the artifacts in the control plane after the WordPress stack has been installed and an example `WordPressInstance` custom resource has been created.
+The diagram shows the result of the following sequence of actions:
+
+1. The user creates an `StackInstall` and specifies the WordPress stack
+1. The Stack Manager installs the CRDs defined in the stack, creates the necessary RBAC primitives, and starts up the WordPress controller in the same namespace as the stack.
+1. The user creates an instance of the `WordPressInstance` custom resource in the same namespace.
+1. The WordPress controller is watching for events on this CRD type in the namespace it is running in.
+1. In response to seeing the creation event for this `WordPressInstance` custom resource, the WordPress controller creates claims for the services that the instance will need, such as a MySQL database and a Kubernetes cluster.
+    1. Note: in order to have permissions to create instances of MySQL and Kubernetes cluster claims, the Wordpress stack will have to have declared those CRD types as dependencies.
```

Comment:
Is there another document that we should be linking to when we talk about dependencies? I am understanding the idea in theory but I am not sure what it would look like or if it's supported yet.

Reply:
No, a design for dependency resolution has not been addressed at all yet, but I'm hoping to do so for the v0.4 release.

Reply:
I would suggest that we add a line to this effect, or some other comment at a top-level which sets expectations for timing . . . I was originally thinking that everything in here was for 0.3.

Reply:
I'm OK with adding language around dependency resolution being out of scope for this design, but I don't want to set timing expectations for features that are out of scope of this doc.  I'd rather that information stay in the roadmap.

Reply:
I will also add a link to the dependency resolution issue (#434) for the user to easily track that way.